core/perf_test/BenchmarkMain.cpp

```diff
@@ -14,7 +14,14 @@
 //
 //@HEADER
 
+// Avoid deprecation warning for ICC
+#ifdef __INTEL_COMPILER
+#pragma warning(disable : 1786)
```

Comment:
I think:
```
#pragma warning(push)
#pragma warning(disable: 1786)
...
#pragma warning(pop)
```
is better.

And do we always want to disable the warning or just with ENABLE_COMPILER_WARNINGS=OFF?